api/capi/src/nnstreamer-capi-single.c

```diff
@@ -1042,24 +1055,23 @@ _ml_single_invoke_internal (ml_single_h single,
      * with the same handle. Thus we can call __invoke without
      * having yet another mutex for __invoke.
      */
-    status = __invoke (single_h);
+    status = __invoke (single_h, single_h->input, single_h->output);
     if (status != ML_ERROR_NONE)
       goto exit;
-    __process_output (single_h);
+    __process_output (single_h, single_h->output);
     single_h->state = IDLE;
   }
 
-  if (single_h->ignore_output == FALSE) {
+exit:
+  if (G_UNLIKELY (status != ML_ERROR_NONE)) {
+    ml_loge ("Failed to invoke the model.");
```

Comment:
Q: at this point (e.g., when `__invoke` was failed), is it okay not to free output data?

Reply:
output handle is destroyed in __invoke() function, at line 246.